inginious/agent/docker_agent/__init__.py

```diff
@@ -459,30 +474,39 @@ async def create_student_container(self, parent_info, socket_id, environment_nam
                     if self._external_ports is not None:
                         for port_to_free in ports:
                             self._external_ports.add(ports[port_to_free])
-                    self._logger.exception("Cannot create student container! No ports are available right now. Please retry later.")
-                    await self._write_to_container_stdin(write_stream, {"type": "run_student_retval", "retval": 254, "socket_id": socket_id})
+                    self._logger.exception(
+                        "Cannot create student container! No ports are available right now. Please retry later.")
+                    await self._write_to_container_stdin(write_stream, {"type": "run_student_retval", "retval": 254,
+                                                                        "socket_id": socket_id})
                     raise CannotCreateJobException('No ports are available right now. Please retry later.')
                 ports[p] = self._external_ports.pop()
 
             try:
                 socket_path = path_join(parent_info.sockets_path, str(socket_id) + ".sock")
+                # get process *NOFILE* resource limits
+                p = psutil.Process()
+                no_file_rlimit = p.rlimit(psutil.RLIMIT_NOFILE)
+                rlimit_per_slot = (
+                    int(no_file_rlimit[0] / self._concurrency), int(no_file_rlimit[1] / self._concurrency))
                 container_id = await self._docker.create_container_student(runtime, environment,
-                                                                           memory_limit, parent_info.student_path, socket_path,
+                                                                           memory_limit, parent_info.student_path,
+                                                                           socket_path,
                                                                            parent_info.systemfiles_path,
                                                                            parent_info.course_common_student_path,
                                                                            parent_info.environment_type,
+                                                                           rlimit_per_slot,
```

Comment:
We should keep this variable clear and call this what it is: file descriptor (fd) limit, or number of file descriptors (nofile). Resource limit (rlimit) is too large and could confuse the reader with the memory limit.